ttnn/cpp/ttnn/operations/full/device/full_device_operation.cpp

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "full_device_operation.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::full {
+void FullOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto any = tensor_args.any;
+    TT_FATAL(any.storage_type() == StorageType::DEVICE, "Full operation error: Any tensor must be on device");
+    TT_FATAL(
+        operation_attributes.memory_config.memory_layout == TensorMemoryLayout::INTERLEAVED,
+        "Full operation error: Not currently supporting sharding");
+    TT_FATAL(
+        operation_attributes.layout == Layout::TILE, "Full operation error: Not currently supporting row major layout");
+
+    const auto shape = operation_attributes.shape;
+
+    TT_FATAL(
+        shape.size() > 1,
```

Comment:
is this limitation connected to bfloat16 requiring at least 2 elements in a row?

Reply:
Since TT doesn't yet support 1D tiled tensors, this code ensures that the user passes a valid shape as a parameter (at least a shape of 2D tensor)